projects/plugins/jetpack/tests/php/media/test-class.jetpack-post-images.php

```diff
@@ -633,4 +633,59 @@ public function test_from_story_block_from_post_id_is_correct_array_videopress_w
 		$expected_poster_url = str_replace( 'mp4', 'jpg', $post_info['img_urls'][1] );
 		$this->assertEquals( $expected_poster_url, $images[1]['src'] );
 	}
+
+	/**
+	 * Test if the array extracted is empty in case post_id is invalid.
+	 *
+	 * @covers Jetpack_PostImages::from_gravatar
+	 * @dataProvider provider_gravatar_invalid_posts
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param int|string|null $post_id  The post ID.
+	 */
+	public function test_from_gravatar_invalid( $post_id ) {
+		$image_details = Jetpack_PostImages::from_gravatar( $post_id );
+		$this->assertEquals( array(), $image_details );
```

Comment:
This could be `assertEmpty`, but I don't think that's a blocker! :) Going to approve this now.